test/Npgsql.Tests/Support/PgPostmasterMock.cs

```diff
@@ -31,6 +31,10 @@ class PgPostmasterMock : IAsyncDisposable
 
         internal string ConnectionString { get; }
 
+        internal bool WaitToBreakOnCancel { get; set; }
+
+        internal TaskCompletionSource WaitToBreakTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Minor nit... It would be nicer if instead of introducing a TCS and flag to delay the cancellation completion, we refactored the mock, breaking Accept apart to multiple method; one would do Accept on the socket ans start reading, return some handle, which we could then "complete".

But this is only a testing infrastructure design nit... I'm completely fine if you don't think it's important enough and prefer to commit as-is.